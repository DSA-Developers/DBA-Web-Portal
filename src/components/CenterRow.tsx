import styled from 'styled-components';
import Row from 'react-bootstrap/Row';

export const CenterRow = styled(Row)`
  display: flex;
  justify-content: center;
`;