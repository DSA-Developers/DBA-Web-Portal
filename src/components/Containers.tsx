import styled from 'styled-components';
import Container from 'react-bootstrap/Container';

export const MainContainer = styled(Container)`
  margin: 0;
  padding: 0;
`;

export const Section = styled.div`
  padding: 5% 20% 5% 20%;
  background-color: #f8f9fb;
  width: 100%;
  text-align: center;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const WhiteSection = styled(Section)`
  background: white;
`;

export const Description = styled.h1`
  font-family: Optima, Segoe, 'Segoe UI', Candara, Calibri, Arial, sans-serif;
  font-size: 35px;
  font-style: normal;
  font-variant: normal;
  font-weight: 100;
  line-height: 60px;
  padding: 5% 10%;
  text-align: center;
`;

export const FullWidth = styled.div`
  bottom: 0;
  width: 100%;
  height: fit-content;
`;