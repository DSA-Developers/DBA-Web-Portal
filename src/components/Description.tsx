import styled from 'styled-components';

export const Description = styled.h1`
  font-family: Optima, Segoe, 'Segoe UI', Candara, Calibri, Arial, sans-serif;
  font-size: 35px;
  font-style: normal;
  font-variant: normal;
  font-weight: 350;
  line-height: 60px;
  padding: 5% 10%;
  text-align: center;
`;