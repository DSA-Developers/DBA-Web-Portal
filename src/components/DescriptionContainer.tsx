import styled from 'styled-components';

export const DescriptionContainer = styled.div`
  padding: 20px 400px 20px 400px;
  background-color: #f8f9fb;
  width: 100%;
  textalign: 'center';
`;