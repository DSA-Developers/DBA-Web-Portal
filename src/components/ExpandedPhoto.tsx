import React from 'react';
import styled from 'styled-components';
import Button from 'react-bootstrap/Button';

const Expand = styled.div`
    background-color: rgba(255,255,255,0.5);
    width: 100vw;
    height: 100vh;
    position: sticky;
    top: 0;
    left: 0;
`

interface Props{};

export const ExpandedPhoto: React.FC<Props> = ({}) => {

return(
    <Expand>
        <Button variant='danger'>
            X
        </Button>

    </Expand>
);

};