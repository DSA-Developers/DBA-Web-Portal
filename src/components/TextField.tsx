import React from 'react';

interface Props {
  text: string;
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const TextField: React.FC<Props> = ({ text, handleChange }) => {
  return (
    <div>
      <input value={text} onChange={handleChange} />
    </div>
  );
}