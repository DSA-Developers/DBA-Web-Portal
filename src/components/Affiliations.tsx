import React from 'react';
import styled from 'styled-components';
import Card from 'react-bootstrap/Card';

import { WhiteSection } from './Containers';
import Title from './Title';
import OrganizationIcon from './OrganizationIcon';

import HSALogo from '../assets/HispanicStudentLogo.png';
import ArSALogo from '../assets/ArgentineStudentsLogo.png';
import ATCLogo from '../assets/ArgentineTango.png';
import CASALogo from '../assets/CubanAmericanLogo.png';
import CALORLogo from '../assets/CentralAmericanLogo.png';
import GSCLogo from '../assets/GatorSalsaLogo.png';
import HCALogo from '../assets/HispanicCommunicatorsLogo.png';
import MASALogo from '../assets/MexicanStudentLogo.png';
import PASSLogo from '../assets/PeruvianAmericanLogo.png';
import PCLogo from '../assets/PorColombiaLogo.png';
import SLLogo from '../assets/SaborLatinoLogo.png';
import UEPALogo from '../assets/UnionDeEstudiantesLogo.png';
import VENSALogo from '../assets/VenezuelanStudentLogo.png';

const clubInfo = [
  { link: 'https://www.ufhsa.com/argentina', photo: ArSALogo, text: 'Argentine Student Association' },
  { link: 'https://www.ufhsa.com/tango', photo: ATCLogo, text: 'Argentine Tango Club' },
  { link: 'https://www.ufhsa.com/cuba', photo: CASALogo, text: 'Cuban American Student Association' },
  { link: 'https://www.ufhsa.com/centralamerica', photo: CALORLogo, text: 'Central American Latino Organization' },
  { link: 'https://www.ufhsa.com/salsa', photo: GSCLogo, text: 'Gator Salsa Club' },
  {
    link: 'https://www.ufhsa.com/hispanic-communicators',
    photo: HCALogo,
    text: 'Hispanic Communicators Association',
  },
  { link: 'https://www.ufhsa.com/mexico', photo: MASALogo, text: 'Mexican-American Student Association' },
  { link: 'https://www.ufhsa.com/peru', photo: PASSLogo, text: 'Peruvian American Student Society' },
  { link: 'https://www.ufhsa.com/colombia', photo: PCLogo, text: 'Por Colombia' },
  { link: 'https://www.ufhsa.com/sabor-latino', photo: SLLogo, text: 'Sabor Latino' },
  {
    link: 'https://www.ufhsa.com/puertorico',
    photo: UEPALogo,
    text: 'La Union de Estudiantes PuertorriqueÃ±os Activos',
  },
  { link: 'https://www.ufhsa.com/venezuela', photo: VENSALogo, text: 'Venezuelan Student Association' },
];

const AffiliationsGrid: React.FC = ({}) => {
  const clubs = clubInfo.map((club) => (
    <OrganizationIcon key={club.link} link={club.link} photo={club.photo} text={club.text} />
  ));

  return (
    <WhiteSection>
      <Title>Parent Organization</Title>
      <br />
      <StyledCard>
        <a href="https://www.ufhsa.com/" target="_blank" style={{ color: '#000' }}>
          <img src={HSALogo} />
          <Card.Title>
            <ParentTitle>Hispanic Student Association</ParentTitle>
          </Card.Title>
        </a>
      </StyledCard>
      <br />
      <Title>
        <h1>In Association With:</h1>
      </Title>
      <Affiliations>{clubs}</Affiliations>
    </WhiteSection>
  );
};

export default AffiliationsGrid;

const ParentTitle = styled.h1`
  font-family: Optima, Segoe, 'Segoe UI', Candara, Calibri, Arial, sans-serif;
  font-size: 27px;
  font-style: normal;
  font-weight: 100;
  line-height: 26.4px;
  padding: 5%;
  text-align: center;
`;

const StyledCard = styled.div`
  transition: transform 0.2s;
  max-width: 250px;
  &:hover {
    transform: scale(1.1);
  }
  margin-bottom: 50px;
`;

const Affiliations = styled.div`
  margin-top: 25px;
  display: grid;
  grid-template-columns: repeat(3, 150px);
  column-gap: 100px;
  row-gap: 100px;
  text-align: center;
`;
