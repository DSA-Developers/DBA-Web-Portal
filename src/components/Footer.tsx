import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import Nav from 'react-bootstrap/Nav';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';

import { FullWidth } from './Containers';

import logoFb from '../assets/facebook.svg';
import logoInst from '../assets/instagram.svg';
import logoGroup from '../assets/groupme.svg';
import DevLogo from '../assets/Dev_Logo.png';

const Footer: React.FC = ({}) => {
  return (
    <FullWidth className="bg-light border-top text-secondary" style={{ position: 'relative' }}>
      <StyledRow>
        <StyledCol>
          <Header>Site Map</Header>
          <Nav className="flex-column">
            <SiteLink as={Link} to="/home">
              Home
            </SiteLink>
            <SiteLink as={Link} to="/about">
              About
            </SiteLink>
            <SiteLink as={Link} to="/gallery">
              Gallery
            </SiteLink>
            <SiteLink as={Link} to="/contact">
              Contact
            </SiteLink>
          </Nav>
        </StyledCol>
        <StyledCol>
          <Header>Connect</Header>
          <SocialMedia>
            <a href="https://www.facebook.com/DSAatUF" target="_blank">
              <Logo src={logoFb}></Logo>
            </a>
            <a href="https://www.instagram.com/uf.dsa/" target="_blank">
              <Logo src={logoInst}></Logo>
            </a>
            <a href="https://groupme.com/join_group/53205471/hXrFzndj" target="_blank">
              <Logo src={logoGroup}></Logo>
            </a>
          </SocialMedia>
        </StyledCol>
        <StyledCol>
          <Header>Contact Us</Header>
          <p>Email: dsaatuf@gmail.com</p>
        </StyledCol>
      </StyledRow>
      <Credits>
        {/* <img src={DevLogo} style={{ width: '8%', height: 'auto' }} /> */}
        <DevelopedBy>
          <em>&ensp;Made with â™¥ by DSA Developers</em>
        </DevelopedBy>
      </Credits>
    </FullWidth>
  );
};

export default Footer;

const StyledRow = styled(Row)`
  display: flex;
  justify-content: center;
  height: fit-content;
  align-items: start;
  margin: 0;
  padding: 10vh 20% 10vh 20%;
`;

const StyledCol = styled(Col)`
  font-size: 1.5rem;
  flex-grow: 1;
  flex-shrink: 2;
  margin: 0 20px 0 20px;
  padding: 0;
  height: fit-content;
`;

const SocialMedia = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 60px);
  gap: 10px;
`;

const Logo = styled.img`
  width: 50px;
  height: 50px;
  &:hover {
    transform: scale(1.2);
    transition-duration: 0.5s;
  }
`;

const Header = styled.h1`
  font-size: 2.5rem;
  margin: 0 0 0.5em 0;
  font-weight: 100;
  text-transform: uppercase;
`;

const SiteLink = styled(Nav.Link)`
  font-size: 1.5rem;
  font-weight: 100;
  color: rgba(0, 0, 0, 0.5);
  :hover {
    color: rgba(0, 0, 0, 0.7);
    text-decoration: none;
  }
  padding: 0 0 0.5em 0;
`;

const Credits = styled(Row)`
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const DevelopedBy = styled.p`
  position: relative;
  font-size: 25px;
  font-weight: 100;
  &:before {
    content: '';
    position: absolute;
    left: -60px;
    bottom: -20%;
    display: block;
    width: 60px;
    height: 55px;
    background-image: url(${DevLogo});
    background-size: contain;
  }
`;
