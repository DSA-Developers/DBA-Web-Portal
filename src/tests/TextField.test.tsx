import renderer from 'react-test-renderer';
import { TextField } from '../components/TextField';

test('Text field created', () => {
  const props = {
    text: 'dsa',
    handleChange: () => {
      return;
    },
  };
  const component = renderer.create(<TextField {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
