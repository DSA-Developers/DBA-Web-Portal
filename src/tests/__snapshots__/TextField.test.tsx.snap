// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text field created 1`] = `
<div>
  <input
    onChange={[Function]}
    value="dsa"
  />
</div>
`;
